modules/s3_hosting/main.tf

```diff
@@ -87,16 +99,17 @@ data "aws_iam_policy_document" "assets_origin" {
 
 # Attach the policy to the bucket
 resource "aws_s3_bucket_policy" "client_assets" {
+  count      = var.create_s3_bucket_policy ? 1 : 0
```

Comment:
Is there a use case where you would want to use an existing bucket but also create a policy? Seems like if the bucket already exists, a policy should also exist.

Reply:
Actually wait, it's probably the other way around.. If you're associating CF with an existing bucket is there any case where it would already have a policy?

Reply:
Oh right, it's just a single policy, you can't append multiple, so you'd need to have a single policy with mulitple principals.

Reply:
yeah, i originally tried to have another variable allowing `extra_s3_bucket_access_principal_arn` 
but then I'd end up with both modules depending on each other. so i just added this option to let people do it manually
```hcl
module s3_hosting "foo" {
  // creating bucket
  // depends on the output of the 2nd s3_hosting module
  extra_s3_bucket_access_principal_arn = module.s3_hosting.bar.asset_access_identity_arn 
}

module s3_hosting "bar" {
  // using bucket foo
  // depends on the bucket to be created
}

Reply:
When you're implementing the real world version of this how are you setting the bucket policy?

Reply:
I had both the module set to `create_s3_bucket_policy` = `false`
then i have the entire policy json / resource  outside with the s3 bucket as data

```
data "aws_s3_bucket" "client_assets" {
  bucket = "deployables.zero-david.online"
}

resource "aws_s3_bucket_policy" "client_assets" {
  depends_on = [module.static_pages_cf_only, module.deployables]
  bucket     = data.aws_s3_bucket.client_assets.id
  policy     = data.aws_iam_policy_document.assets_origin.json
}

data "aws_iam_policy_document" "assets_origin" {
  statement {
    actions   = ["s3:GetObject"]
    resources = ["arn:aws:s3:::${data.aws_s3_bucket.client_assets.id}/*"]

    principals {
      type        = "AWS"
      identifiers = [module.static_pages_cf_only.cf_origin_assets_access_identity_arn, module.deployables.cf_origin_assets_access_identity_arn]
    }
  }

  statement {
    actions   = ["s3:ListBucket"]
    resources = ["arn:aws:s3:::${data.aws_s3_bucket.client_assets.id}"]

    principals {
      type        = "AWS"
      identifiers = [module.static_pages_cf_only.cf_origin_assets_access_identity_arn, module.deployables.cf_origin_assets_access_identity_arn]
    }
  }
}
```

Reply:
Should be fine, though we may need some documentation around it since it happens outside the modules. Also you probably want to add a `depends_on` to your `aws_s3_bucket` data source.